### PR TITLE
[Feature] lake table support multi worker schema change

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
Signed-off-by: starrocks-xupeng <xupeng@starrocks.com>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 12 (41.67%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|1|8|12.50%|[329, 330, 331, 332, 333, 334, 335]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|1|1|100.00%|[]|

**Review by sduzh — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 1]
```diff
@@ -322,6 +325,15 @@ public List<Long> createShards(int numShards, int replicaNum, FilePathInfo pathI
 
             for (int i = 0; i < numShards; ++i) {
                 builder.setShardId(GlobalStateMgr.getCurrentState().getNextId());
+                if (matchShardIds != null) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:328]: should check `matchShardIds.size() == numShards`

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java [hunk 2]
```diff
@@ -307,7 +309,8 @@ public List<ShardGroupInfo> listShardGroup() {
         }
     }
 
-    public List<Long> createShards(int numShards, int replicaNum, FilePathInfo pathInfo, FileCacheInfo cacheInfo, long groupId)
+    public List<Long> createShards(int numShards, int replicaNum, FilePathInfo pathInfo, FileCacheInfo cacheInfo, long groupId,
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:312]: how about add a new method createShards() without the matchShardIds parameter which is preferred in most cases. So most calls will still go without the `matchShardIds` parameter and won't have to pass a `null`  for it.

Comment (starrocks-xupeng) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:317]: done

File: fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java (hunk 1 above)
Comment (starrocks-xupeng) [reply fe/fe-core/src/main/java/com/starrocks/lake/StarOSAgent.java:336]: done

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by starrocks-xupeng** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 6 / 14 (42.86%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/StarOSAgent.java|3|11|27.27%|[321, 337, 338, 339, 340, 341, 342, 343]|
|:large_blue_circle:|com/starrocks/alter/LakeTableAlterJobV2Builder.java|3|3|100.00%|[]|

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**